src/Controller/Component/DataTablesComponent.php

```diff
@@ -206,4 +251,41 @@ private function _addCondition($column, $value, $type = 'and')
             $this->config('matching', [$association => $condition]); // merges
         }
     }
+
+    /**
+     * Get comparison by entity and column name.
+     *
+     * @param  string $column Database column
+     * @return int            Database comparison type
+     */
+    protected function getComparison($column)
+    {
+        $config = new Collection($this->config('comparison'));
+        $tableName = $entity = $this->_tableName;
+
+        // Attempt to find the table name
+        if (false !== ($pos = strpos($column, '.'))) {
+            $entity = substr($column, 0, $pos);
+            $tableName = TableRegistry::get($entity)->table();
+            $columnName = substr($column, $pos + 1);
+        }
+
+        // See if the controller defined a specific comparison type for this field
+        $userConfig = $config->filter(function ($item, $key) use ($entity, $columnName) {
+            return strtolower($key) === strtolower(sprintf('%s.%s', $entity, $columnName));
+        });
+
+        if (!$userConfig->isEmpty()) {
+            return $userConfig->first();
+        }
+
+        // Check application config if the column type has a default comparison defined
+        $columnDesc = $this->_collection->describe($tableName)->column($columnName);
+
+        if(isset($this->_defaultComparison[$columnDesc['type']])) {
```

Comment:
I would prefer `return $this->_defaultComparison[$columnDesc['type']] ?? '='` here instead of the if.
As noted where the method is used, a string is expected.